src/categorical_algebra/FinSets.jl

```diff
@@ -36,8 +37,11 @@ functions, through types `FinSet{S} where S <: AbstractSet`, as well as the
 skeleton of this category, through the type `FinSet{Int}`. In the latter case,
 the object `FinSet(n)` represents the set {1,...,n}.
 """
-@auto_hash_equals struct FinSet{S} <: AbstractSetOb{S}
+@auto_hash_equals struct FinSet{S,T} <: AbstractSetOb{S,T}
   set::S
+  function FinSet(i::Int)
```

Comment:
This constructor should probably be outside the `struct` so we can construct other kinds of `FinSet`, something like:

```julia
FinSet(n::Int) = FinSet{Int,Int}(n)
FinSet(set::S) where {T,S<:AbstractSet{T}} = FinSet{S,T}(set)
```